packages/edit-site/src/components/page-patterns/index.js

```diff
@@ -325,6 +325,7 @@ export default function DataviewsPatterns() {
 				filterBy: {
 					operators: [ OPERATOR_IN ],
 					isPrimary: true,
+					singleSelection: true,
```

Comment:
I feel like we should have the single selection by default, similar for example to `select` tag with the `multiple` attribute.

Reply:
I started that way, but then looked at the current uses cases:

- Pages. Both Author and Status should be multiselection.
- Patterns. Sync should be single selection.
- Templates. Author filter should be multiselection.
- Parts. Author filter should be multiselection.

All filters but one (sync in patterns) use multiselection, so I thought we should actually optimize for the common use case.

Reply:
In general I think this will probably be inferred from the new operators.

Reply:
> In general I think this will probably be inferred from the new operators.

Can you expand this comment?

Reply:
Instead of the `singleSelection` flag we could use a flag to declare what are the supported relationship? As in:

```js
relationship: [ ANY_OF, ALL_OF ]
```

Reply:
> `relationship: [ ANY_OF, ALL_OF ]`

We'll need something like this for category and tag filtering. E.g. It should be possible to; view posts in Category A _or_ Category B. Or: View posts in Category A _and_ Category B. The inverse should also be possible (posts _not in_ Category A or B).

The UI should provide a way for users to change this operator in the UI.

Reply:
Pushed a change for this and updated the docs to explain:

> Allowed operators for fields of type `enumeration`:
> 
> - `equal`: whether an item is equal to a single value.
> - `notEqual`: whether an item is not equal to a single value.
> - `in`: whether an item is in a list of values.
> - `notIn`: whether an item is not in a list of values.
>
> By default, a field of type `enumeration` supports `in` and `notIn` operators — this is, it supports multiselection and negation.
> 
> A filter cannot mix single-selection (`equal`, `notEqual`) & multi-selection (`in`, `notIn`) operators. If a single-selection operator is present in the list of valid operators, the multi-selection ones will be discarded and the filter won't allow selecting more than one item.

Reply:
Examples:

- Single-selection filter: `operators: [ OPERATOR_EQUAL, OPERATOR_NOT_EQUAL ]`
- Multi-selection filter (the default): `operators: [ OPERATOR_IN, OPERATOR_NOT_IN ]`
- Single-selection filter that does not support negation: `operators: [ OPERATOR_EQUAL ]`
- etc.

This PR only adds the single-selection. The existing multi-selection filters (in/not in) are `OR` operators. I'll add the `AND` operator in a follow-up.

For reference, in terms of [nomenclature](https://github.com/WordPress/gutenberg/pull/59610/files#diff-745eee943d16435ef150be06d0022126d2acf9409d0c50e8cc25fb2699575073R33) we'd have (and their negations):

| Operator | Key | Label | Example |
| --- | --- | --- | --- |
| OPERATOR_EQUALS | `equal-filter` | `is` | Author is Admin |
| OPERATOR_IN | `in-filter` | `in` | Author in Admin, Editor |
| OPERATOR_ALL | `all-filter` | `is all` | Tags is all of Books, English, School |